genetic_algorithms/problems/traveling_salesman_problem/visualization.py

```diff
@@ -0,0 +1,45 @@
+import pygame, sys
+from genetic_algorithms.algorithm_wrappers.algorithm_wrapper import AlgorithmWrapper
+from genetic_algorithms.problems.traveling_salesman_problem.problem_model import TravelingSalesmanModel
+from genetic_algorithms.problems.traveling_salesman_problem.state import TravelingSalesmanState
+
+class Visualization(AlgorithmWrapper):
+    def _perform_side_effects(self, model: TravelingSalesmanModel, state: TravelingSalesmanState):
+        pygame.init()
+        self.screen = pygame.display.set_mode((900, 900))
+        _points = model.points
+        _route = state.route
+        self.find_extreme(_points)
+        _route_points = [(self.scale(_points[idx])) for idx in _route]
+
```

Comment:
Jak pisałem wyżej, dolne podkreślenie - to konwencja w Pythonie używana do tego, żeby pokazać, że funkcja jest "prywatna". (w cudzysłowiach, ponieważ w pythonie nie ma czegoś takiego jak prywatne metody)
Natomiast zmienne lokalne raczej nie powinny być tak oznaczane.